Common/Global/GlobalVanillaItem.cs

```diff
@@ -3,39 +3,40 @@
 using Terraria.DataStructures;
 using Terraria.ID;
 using Terraria.ModLoader;
+using System;
+using System.Collections.Generic;
 
 namespace TheBereftSouls.Common.Global
 {
 	public class GlobalVanillaItem : GlobalItem
 	{
 		//Untested - JIT used to prevent crashes if Calamity isn't enabled.
+		
+
+		// Changes following Vanilla Weapons To Rogue, uses Integer due to how Vanilla ID work
+		// In order its Bone Javelin, Throwing Knife, Shuriken and Bone Dagger
+		private static List<int> Vanilla_Weapons_To_Rogue = new List<int>
+		{
+			3378,
+			279,
+			42,
+			3379
+		};
```

Comment:
Also, you used a mix-up between `PascalCase` and `snake_case` as the naming convention everywhere. Not that there's something inherently wrong with it (although it doesn't follow the language convention which is pretty inconvenient). The naming convention for fields is `camelCase` and the naming convention for `const` fields is `PascalCase`.

[.NET Style Guidelines](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md) (point 3 for fields and point 12 for constants)

Reply:
I guess I could make a PR with a CI/CD pipeline and an `.editorconfig` file to catch those style errors. It'd be nice to be able to apply them on the spot.

Reply:
That might be nice since its always better to catch stuff like that earlier than later.

Reply:
CI/CD is impossible since it relies on having Terraria installed and having the game files public which is illegal I'm pretty sure. Managed to use a code analysis tool though so it's all good! I won't open my PR since it may have conflicts with your files but you can check the [branch from my fork](https://github.com/1Git2Clone/TheBereftSouls/tree/features/code-analysis) if you're interested.

What's left is to make a `CONTRIBUTING.md` file but that deserves a different branch and PR related to it. Basically you can format with `dotnet format` and you need to check your code style with `dotnet build --no-incremental`.